src/java/boa/compiler/visitors/AbstractVisitorNoArgNoRet.java

```diff
@@ -32,6 +32,23 @@
 public abstract class AbstractVisitorNoArgNoRet {
 	protected void initialize() { }
 
+	public Node currentNode = null;
+
+	public void dfs(final Node node, java.util.HashMap<Integer,String> nodeVisitStatus) {
+		currentNode = node;
+		nodeVisitStatus.put(node.nodeId,"visited");
+		cfgAnalysis(node);
+		for (Node succ : node.successors) {
+		    if (nodeVisitStatus.get(succ.nodeId).equals("unvisited")) {
+			dfs(succ, nodeVisitStatus);
+		    }
+		}
+	}
+
+	public void cfgAnalysis(final Node node) {
```

Comment:
Is cfgAnalysis method similar to start method? Looks like a duplicate to me.

Reply:
Seems like it isn't used anywhere so I'll remove it.  Same with that 'currentNode' field.